File: EDOs con condiciones iniciales/euler.f90
!------------------------------------------------------------------------
!EDOs con condicion inicial mediante el metodo de Euler hacia adelante
!Autor: Alvaro Siesquen
!------------------------------------------------------------------------
program euler
implicit none
integer i,N
real h, t, y
real f

open(8,file='data_euler.dat', status='unknown')		!Archivo externo para guardar los datos

!Procedimiento
h = 0.001		!Paso
N = 10000		!Numero de puntos

print*,'t inicial'
read*,t
print*,'y inicial'
read*,y

do i=1,N+1,1
	write(8,*)t,y
    y = y + h*f(y,t)
	t = t+h
end do
print*,'t transcurrido:',h*N
print*,'Finalizado'
end program

real function f(y,t)
real t,y
    f = exp(t)
end function

File: EDOs con condiciones iniciales/euler.py
# ------------------------------------------------------------------------
# EDOs con condicion inicial mediante el metodo de Euler hacia adelante
# Autor: Alvaro Siesquen
# ------------------------------------------------------------------------

import numpy as np
f = lambda y,t: np.exp(t)

def met_euler(y, h, t, tf, f):
    N = int((tf-t)/h)
    Y = [y]
    T = [t]

    for i in range(N):
        y += h*f(y,t)
        t += h
        Y.append(y)
        T.append(t)
    return Y,T